### ISSUE TITLE
XML output - "bodyend" inside location tag sometimes prints the wrong row. (Origin: bugzilla #741992)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#5695]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.8.8* on platform *Other*
Assigned to: Dimitri van Heesch

On 2014-12-26 10:59:11 +0000, erez.oxman@gmail.com wrote:

    I've noticed it on several projects on Python, that sometimes the "body end"
    misses for kind="class" (I've only found it on classes, not on functions).
    For example, on Bleach:
    (https://github.com/jsocol/bleach/tree/master/bleach):
    
    class BleachSanitizerMixin(HTMLSanitizerMixin) which appears at
    https://github.com/jsocol/bleach/blob/master/bleach/sanitizer.py#L14
    ends at maximum line 133-135. The next class in that file starts at 136.
    As you can see in the following output,the "bodyend" line is 136.
    I can send more examples if you need.
    
```    
      </compounddef><compounddef xmlns:xsi="http://www.w3.org/2001/XMLSchema-
    instance" id="classbleach_1_1sanitizer_1_1BleachSanitizerMixin" kind="class"
    language="Python" prot="public">
        <compoundname>bleach::sanitizer::BleachSanitizerMixin</compoundname>
        <basecompoundref prot="public" virt="non-
    virtual">HTMLSanitizerMixin</basecompoundref>
        <derivedcompoundref refid="classbleach_1_1sanitizer_1_1BleachSanitizer"
    prot="public" virt="non-
    virtual">bleach.sanitizer.BleachSanitizer</derivedcompoundref>
          <sectiondef kind="public-static-attrib">
          <memberdef kind="variable"
    id="classbleach_1_1sanitizer_1_1BleachSanitizerMixin_1ac7d112b733510ee2fa13b8fe372137ba"
    prot="public" static="yes" mutable="no">
            <type>list</type>
            <definition>list
    bleach.sanitizer.BleachSanitizerMixin::allowed_svg_properties</definition>
            <argsstring/>
            <name>allowed_svg_properties</name>
            <initializer>= []</initializer>
            <briefdescription>
            </briefdescription>
            <detaileddescription>
            </detaileddescription>
            <inbodydescription>
            </inbodydescription>
            <location
    file="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    line="17" column="1"
    bodyfile="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    bodystart="17" bodyend="-1"/>
          </memberdef>
          </sectiondef>
          <sectiondef kind="public-attrib">
          <memberdef kind="variable"
    id="classbleach_1_1sanitizer_1_1BleachSanitizerMixin_1a99de9955bee796ca9a8c2a29e3452d25"
    prot="public" static="no" mutable="no">
            <type/>
<definition>bleach.sanitizer.BleachSanitizerMixin::wildcard_attributes</definition>
            <argsstring/>
            <name>wildcard_attributes</name>
            <briefdescription>
            </briefdescription>
            <detaileddescription>
            </detaileddescription>
            <inbodydescription>
            </inbodydescription>
            <location
    file="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    line="33" column="1"
    bodyfile="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    bodystart="33" bodyend="-1"/>
          </memberdef>
          </sectiondef>
          <sectiondef kind="public-func">
          <memberdef kind="function"
    id="classbleach_1_1sanitizer_1_1BleachSanitizerMixin_1a9bd815e16606d04b3dc72368e65d852e"
    prot="public" static="no" const="no" explicit="no" inline="no" virt="non-
    virtual">
            <type>def</type>
            <definition>def
    bleach.sanitizer.BleachSanitizerMixin.sanitize_token</definition>
            <argsstring>(self, token)</argsstring>
            <name>sanitize_token</name>
            <param>
              <type>self</type>
              <defname>self</defname>
            </param>
            <param>
              <type>token</type>
              <defname>token</defname>
            </param>
            <briefdescription>
            </briefdescription>
            <detaileddescription>
    <para><verbatim>Sanitize a token either by HTML-encoding or dropping.

    Unlike HTMLSanitizerMixin.sanitize_token, allowed_attributes can be
    a dict of {'tag': ['attribute', 'pairs'], 'tag': callable}.
    
    Here callable is a function with two arguments of attribute name
    and value. It should return true of false.
    
    Also gives the option to strip tags instead of encoding.</verbatim> </para>
    </detaileddescription>
            <inbodydescription>
            </inbodydescription>
            <location
    file="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    line="19" column="1"
    bodyfile="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    bodystart="19" bodyend="99"/>
          </memberdef>
          <memberdef kind="function"
    id="classbleach_1_1sanitizer_1_1BleachSanitizerMixin_1ab13d5bd6846ea56941d5e18a7b943d9b"
    prot="public" static="no" const="no" explicit="no" inline="no" virt="non-
    virtual">
            <type>def</type>
            <definition>def
    bleach.sanitizer.BleachSanitizerMixin.sanitize_css</definition>
            <argsstring>(self, style)</argsstring>
            <name>sanitize_css</name>
            <param>
              <type>self</type>
              <defname>self</defname>
            </param>
            <param>
              <type>style</type>
              <defname>style</defname>
            </param>
            <briefdescription>
            </briefdescription>
            <detaileddescription>
    <para><verbatim>HTMLSanitizerMixin.sanitize_css replacement.
    
    HTMLSanitizerMixin.sanitize_css always whitelists background-*,
    border-*, margin-*, and padding-*. We only whitelist what's in
    the whitelist.</verbatim> </para>        </detaileddescription>
            <inbodydescription>
            </inbodydescription>
            <location
    file="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    line="100" column="1"
    bodyfile="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    bodystart="100" bodyend="135"/>
          </memberdef>
          </sectiondef>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
    <para><verbatim>Mixin to replace sanitize_token() and
    sanitize_css().</verbatim> </para>    </detaileddescription>
        <inheritancegraph>
          <node id="8">
            <label>bleach.sanitizer.BleachSanitizerMixin</label>
            <link refid="classbleach_1_1sanitizer_1_1BleachSanitizerMixin"/>
            <childnode refid="9" relation="public-inheritance">
            </childnode>
          </node>
          <node id="10">
            <label>bleach.sanitizer.BleachSanitizer</label>
            <link refid="classbleach_1_1sanitizer_1_1BleachSanitizer"/>
            <childnode refid="8" relation="public-inheritance">
            </childnode>
          </node>
          <node id="9">
            <label>HTMLSanitizerMixin</label>
          </node>
        </inheritancegraph>
        <collaborationgraph>
          <node id="11">
            <label>bleach.sanitizer.BleachSanitizerMixin</label>
            <link refid="classbleach_1_1sanitizer_1_1BleachSanitizerMixin"/>
            <childnode refid="12" relation="public-inheritance">
            </childnode>
          </node>
          <node id="12">
            <label>HTMLSanitizerMixin</label>
          </node>
        </collaborationgraph>
        <location
    file="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    line="14" column="1"
    bodyfile="/home/erez/project/temp_unzipped/2/python/bleach/sanitizer.py"
    bodystart="14" bodyend="136"/>
```

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
